NDTensors/src/dense/generic_array_constructors.jl

```diff
@@ -0,0 +1,33 @@
+using .TypeParameterAccessors:
+  default_type_parameter,
+  parenttype,
+  set_eltype,
+  specify_default_type_parameters,
+  type_parameter
+##TODO replace randn in ITensors with generic_randn
+## and replace zeros with generic_zeros
+
+# This is a file to write generic fills for NDTensors.
+#  This includes random fills, zeros, ...
+
+function generic_randn(
+  StoreT::Type{<:Dense}, dims::Tuple{Integer}; rng=Random.default_rng()
+)
+  StoreT = specify_default_type_parameters(StoreT)
+  DataT = specify_type_parameter(type_parameter(StoreT, parenttype), eltype, eltype(StoreT))
+  @assert eltype(StoreT) == eltype(DataT)
+
+  data = generic_randn(DataT, dims; rng=rng)
+  StoreT = set_datatype(StoreT, typeof(data))
+  return StoreT(data)
+end
+
+function generic_zeros(StoreT::Type{<:Dense}, dims::Tuple{Integer})
```

Comment:
```suggestion
function generic_zeros(StoreT::Type{<:Dense}, dims::Integer)
```

Reply:
My proposal was to define both of them in terms of `::Integer`, not `::Tuple{Integer}`, since it is a simpler interface for new types to overload in general.

Reply:
If you define with integer this code fails.
 ```
julia> NDTensors.generic_zeros(Dense, (2,))
ERROR: Not implemented
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] datatype(#unused#::Type{Dense})
   @ NDTensors ~/.julia/dev/ITensors/NDTensors/src/tensorstorage/tensorstorage.jl:5
 [3] parenttype(storagetype::Type{Dense})
   @ NDTensors ~/.julia/dev/ITensors/NDTensors/src/tensor/set_types.jl:26
 [4] is_wrapped_array(arraytype::Type{Dense})
   @ NDTensors.TypeParameterAccessors ~/.julia/dev/ITensors/NDTensors/src/lib/TypeParameterAccessors/src/base/abstractarray.jl:31
 [5] trait(#unused#::Type{NDTensors.TypeParameterAccessors.IsWrappedArray{Dense}})
   @ NDTensors.TypeParameterAccessors ~/.julia/packages/SimpleTraits/l1ZsK/src/SimpleTraits.jl:171
 [6] unwrap_array_type
   @ ~/.julia/packages/SimpleTraits/l1ZsK/src/SimpleTraits.jl:331 [inlined]
 [7] generic_zeros(arraytype::Type{Dense}, dims::Tuple{Int64})
   @ NDTensors ~/.julia/dev/ITensors/NDTensors/src/abstractarray/generic_array_constructors.jl:25
 [8] top-level scope
   @ REPL[2]:1
 ```

Reply:
My only concern is that if you do something like this `generic_zeros(Dense, (2,))` it calls the `AbstractArray` version of this funciton

Reply:
Right, so then you should rewrite the `AbstractArray` version so that doesn't happen.

Reply:
I am not sure I understand what you're suggesting to do. I reverted the code back the original design you suggested which works for `generic_zeros(Dense, 2)` and `generic_zeros(Dense, (2,))`